docs/DeveloperGuide.md

```diff
@@ -1,38 +1,254 @@
 # Developer Guide
 
-## Acknowledgements
+## Introduction
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+Budget Buddy aims to help those who want a Command Line Interface App that is a one-stop solution to keep track of their
+expenses, help them with budgeting and also be help them to visualise what they have spent.
 
-## Design & implementation
+Overspending and the lack of savings is a significant problem found in most fresh graduates. We want to create tools
+necessary for them so that they can realize their actual spending and achieve financial freedom in the future.
+
+## Setting up the project
+
+## Design
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Main Component
+
+![MainSeq.png](uml/MainSequence.png)
```

Comment:
<img width="415" alt="image" src="https://user-images.githubusercontent.com/81511355/228759573-1035eeb3-2811-4f27-a3f7-6b9c34a7768f.png">
Ensure that you write the Class contructor in the sequence diagram.

e.g.
<img width="332" alt="image" src="https://user-images.githubusercontent.com/81511355/228759867-eca0bf0a-34ab-466f-a0b1-3c9058ab84fb.png">